### PR TITLE
Dtspo4185 fix azuread errors

### PR DESCRIPTION
**Pull request by darren221980**
### Change description ###

As part of DTSPO4185 i noticed that due to azuread moving to v 2.0.1 that the name field was no longer accepted and has changed to display_name.

name field updated to use display_name.

Had to add locals so that the genesis module will work with cft ptlsbox environment.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

**Review by naikajah — APPROVED** [review]

File: 03-interpolated-defaults.tf [hunk 1]
```diff
@@ -1,11 +1,20 @@
+locals {
+  
+  sub_env = var.environment == "ptlsbox" ? "ptl-sbox" : var.environment
+  aad_group_name = contains(["ptlsbox", "ptl"], var.environment) ? "DTS Contributors (sub:dts-sharedservices${local.sub_env})" : "DTS Contributors (sub:dts-sharedservices-${local.sub_env})"
```
Comment (thomast1906) [inline 03-interpolated-defaults.tf:4]: This isn't correct - if we were to deploy sbox from CFT?

Comment (darren221980) [reply 03-interpolated-defaults.tf:4]: I've ran a plan against SBOX env using this branch here and all looks good - https://dev.azure.com/hmcts/CNP/_build/results?buildId=170521&view=results

Comment (thomast1906) [reply 03-interpolated-defaults.tf:4]: As discussed - this needs to be CFT specific

Comment (thomast1906) [reply 03-interpolated-defaults.tf:4]: As discussed, will resolved for now

**Review by thomast1906 — APPROVED** [review]

**Comment by thomast1906** [comment]
@timja @TomKimber unless we want to re-run the bootstrapping for ptlsbox & ptl? The AD groups have been created without "-"

**Comment by timja** [comment]
@msl8r is more familiar with this, / he has time planned to look at this ASAP

**Comment by msl8r** [comment]
We could put the workaround in here: https://github.com/hmcts/ops-bootstrap/blob/a84b36010a9c1f0afca085214d7d47520d0287bf/Terraform/Modules/Infrastructure/Management/main.tf#L79-L87

But I think doing it this way may be the better option imo, if someone is looking for the group it matches up with the subscription name.

**Review by timja — APPROVED** [review]

**Review by msl8r — APPROVED** [review]

**Pull request merged**